utils.js

```diff
@@ -1,7 +1,16 @@
 const arrayToObject = (header, data) => {
-  return data.reduce((acc, crt) => {
+  return data.reduce((acc, crt, index) => {
+    if (!crt) {
+      console.error(`Encountered undefined data row at index ${index}`);
```

Comment:
Validation should be done outside the function, keeping it pure and only callable when all conditions ar emet